### PR TITLE
Implement member import with tier

### PR DESCRIPTION
**Pull request by mike182uk**
refs https://github.com/TryGhost/Product/issues/3629

### CODE REVIEW

File: ghost/members-importer/lib/MembersCSVImporter.js [hunk 1]
```diff
@@ -33,24 +34,27 @@ module.exports = class MembersCSVImporter {
...
         this._getMembersRepository = getMembersRepository;
         this._getDefaultTier = getDefaultTier;
+        this._getTierByName = getTierByName;
         this._sendEmail = sendEmail;
         this._isSet = isSet;
         this._addJob = addJob;
         this._knex = knex;
         this._urlFor = urlFor;
         this._context = context;
+        this._tierCache = new Map();
```
Comment (allouis) [inline ghost/members-importer/lib/MembersCSVImporter.js]: I can't see that we ever clear this cache - which means we could be dealing with outdated values.

File: ghost/members-api/lib/repositories/MemberRepository.js [hunk 2]
```diff
@@ -1635,4 +1639,162 @@ module.exports = class MemberRepository {
...
+     * This method should be used in-conjunction with the `linkSubscription` method to ensure that the changes
+     * made in Stripe are reflected in Ghost - This is not executed as part of this to allow for flexibility
+     *
+     * @param {Object} data
+     * @param {String} data.customer_id - Stripe customer ID
+     * @param {String} data.product_id - Ghost product ID
+     * @param {Object} options
+     * @returns {Promise<Object>}
+     */
+    async forceStripeSubscriptionToProduct(data, options) {
```
Comment (allouis) [inline ghost/members-api/lib/repositories/MemberRepository.js:1668]: I don't think this method belongs in here - we want to decouple the `members-api` package from Stripe completely, and this takes us in the other direction.

I would consider moving this into the import package, or a separate one - but I think importer is fine for now

**Review by allouis — COMMENTED** [review]
Approach looks good! Would be great if we can move the new stripe code out of members-api though, that will allow us to keep members/stripe decoupled as well as not have to return the archive method

File: ghost/members-importer/lib/MembersCSVImporter.js (hunk 1 above)
Comment (mike182uk) [reply ghost/members-importer/lib/MembersCSVImporter.js]: [It gets cleared on each execution ](https://github.com/TryGhost/Ghost/pull/17506/files#diff-a34d8cdc0a7caff73fbddfd0f4e205fb9b247d1b4e1b8b648c05f40a29812a79R121-R122) 🙂

File: ghost/members-api/lib/repositories/MemberRepository.js (hunk 2 above)
Comment (mike182uk) [reply ghost/members-api/lib/repositories/MemberRepository.js:1668]: Ok cool , I'll look at moving this into the `members-importer` package seeming though that this logic will probably only ever be called within the context of importing members 👍

Comment (mike182uk) [reply ghost/members-api/lib/repositories/MemberRepository.js]: I've moved this logic over to the `members-importer` package now - ended up being a little trickier than expected to get the relevant dependencies passed around so had to rejig the package a little

**Pull request merged**